src/main/java/competition/simulation/MapleSimulator.java

```diff
@@ -0,0 +1,132 @@
+package competition.simulation;
+
+import competition.Robot;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.measure.Distance;
+import org.ironmaple.simulation.drivesims.COTS;
+import xbot.common.advantage.AKitLogger;
+import xbot.common.controls.sensors.mock_adapters.MockGyro;
+import xbot.common.logic.TimeStableValidator;
+
+import static edu.wpi.first.units.Units.Meters;
+import static edu.wpi.first.units.Units.Seconds;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.ironmaple.simulation.SimulatedArena;
+import org.ironmaple.simulation.drivesims.SelfControlledSwerveDriveSimulation;
+import org.ironmaple.simulation.drivesims.SwerveDriveSimulation;
+import org.ironmaple.simulation.drivesims.configs.DriveTrainSimulationConfig;
+
+@Singleton
+public class MapleSimulator implements BaseSimulator {
```

Comment:
⭐ if we're going to keep putting more stuff in here, should we just call this Simulator? Or BaseSwerveSimulator?